src/minisatip.c

```diff
@@ -1939,7 +1948,8 @@ int main(int argc, char *argv[]) {
                           (socket_action)ssdp_discovery);
         if (si < 0 || si1 < 0)
             FAIL("sockets_add failed for ssdp");
-    }
+    } else
+        generate_uuid();
```

Comment:
Is this needed? Would it be possible to call `generate_uuid()` just in one place?

Reply:
Yes, it's necessary because the **global** var `uuidi` that controls the UUID generation. So we need to be sure that _in all cases_ the UUID is created. So for reason and in order to not generate side effects errors if the code changes in the future, then it's preferable to call to the new `generate_uuid()` function every time a SSDP call is done and overcomed by the `opts.disable_ssdp` option. Therefore it's more safe to add this line here.

Please check it and merge! 😉

Reply:
I still think it would be preferable to all `generate_uuid()` just once sometime during startup. Then we could even get rid of `uuidi` completely.

Reply:
Hi @Jalle19 ,

That's my original idea. However, I'm not the one that has introduced the `uuidi` var. This var has _specifically_ the objective of caching the creation of the UUID. I feel this was introduced (by @catalinii) to overcome potential errors of different edge cases or to permit to recalculate it. Anyway the fix is clean, simple and it works, so I suggest to merge it.

Regards.

Reply:
Hi @Jalle19 ,

I undertand you suggestion. And I agree with the idea of some refactor. However, now I don't have much time. So because the code is clean and it fixes the bug, why not merge it now and in the future someone will refactor it?

Reply:
I'll see if I can find the time to do the necessary changes

Reply:
Hi @catalinii ,

Can you comment please?

Reply:
A lot of things have been done historically for reasons probably they are no longer valid.
Cleaning the code is the best way forward IMHO, so I think that stripping the code out from there and putting it in the initialization part would make the code more clean and it would be called just once.

Reply:
Hi @catalinii ,

Thank you for the response. I agree with you and @Jalle19 . I prefer it the code will be cleaned. However, I don't have time now to do it. So I hope @Jalle19 could enhance the PR. In fact it's sufficient to check _all_ calls to `generate_uuid()`. The problem in my case it's the time necessary to do an extensible check of all edge cases. And for this reason I've suggested to merge the PR and _after_ refactor the code. Perhaps the most simple way is to encapsulate all initialization tasks inside a new function and execute the call to `generate_uuid()` inside it.

Regards.